src/commands.rs

```diff
@@ -0,0 +1,1173 @@
+        for p in paths {
+            res = Some(match op {
+                NumOp::INCR => redis_key.incr_by(p, &number)?,
+                NumOp::MULT => redis_key.mult_by(p, &number)?,
+                NumOp::POW => redis_key.pow_by(p, &number)?,
+            });
+        }
+        redis_key.apply_changes(ctx, cmd)?;
+        Ok(res.unwrap().to_string().into())
+    } else {
+        Err(RedisError::String(format!(
+            "Path '{}' does not exist or does not contains a number",
+            path
+        )))
+    }
+}
+
+pub fn command_json_num_incrby<M: Manager>(
+    manager: M,
+    ctx: &Context,
+    args: Vec<String>,
+) -> RedisResult {
+    command_json_num_op(manager, ctx, args, "json.numincrby", NumOp::INCR)
+}
+
+pub fn command_json_num_multby<M: Manager>(
+    manager: M,
+    ctx: &Context,
+    args: Vec<String>,
+) -> RedisResult {
+    command_json_num_op(manager, ctx, args, "json.nummultby", NumOp::MULT)
+}
+
+pub fn command_json_num_powby<M: Manager>(
+    manager: M,
+    ctx: &Context,
+    args: Vec<String>,
+) -> RedisResult {
+    command_json_num_op(manager, ctx, args, "json.numpowby", NumOp::POW)
+}
+
+pub fn command_json_bool_toggle<M: Manager>(
+    manager: M,
+    ctx: &Context,
+    args: Vec<String>,
+) -> RedisResult {
+    let mut args = args.into_iter().skip(1);
+    let key = args.next_string()?;
+    let path = backwards_compat_path(args.next_string()?);
+    let mut redis_key = manager.open_key_write(ctx, &key)?;
+
+    let root = redis_key
+        .get_value()?
+        .ok_or_else(RedisError::nonexistent_key)?;
+    let paths = find_paths(&path, root, |v| v.get_type() == SelectValueType::Bool)?;
+    if paths.len() > 0 {
+        let mut res = None;
+        for p in paths {
+            res = Some(redis_key.bool_toggle(p)?);
+        }
+        redis_key.apply_changes(ctx, "json.toggle")?;
+        Ok(res.unwrap().to_string().into())
+    } else {
+        Err(RedisError::String(format!(
+            "Path '{}' does not exist",
+            path
+        )))
+    }
+}
+
+pub fn command_json_str_append<M: Manager>(
+    manager: M,
+    ctx: &Context,
+    args: Vec<String>,
+) -> RedisResult {
+    let mut args = args.into_iter().skip(1);
+
+    let key = args.next_string()?;
+    let path_or_json = args.next_string()?;
+
+    let path;
+    let json;
+
+    // path is optional
+    if let Ok(val) = args.next_string() {
+        path = backwards_compat_path(path_or_json);
+        json = val;
+    } else {
+        path = JSON_ROOT_PATH.to_string();
+        json = path_or_json;
+    }
+
+    let mut redis_key = manager.open_key_write(ctx, &key)?;
+
+    let root = redis_key
+        .get_value()?
+        .ok_or_else(RedisError::nonexistent_key)?;
+
+    let paths = find_paths(&path, root, |v| v.get_type() == SelectValueType::String)?;
+    if paths.len() > 0 {
+        let mut res = None;
+        for p in paths {
+            res = Some(redis_key.str_append(p, json.clone())?);
```

Comment:
- Will not report on paths leading to non-String values (different behavior than before)

- Might modify only a sub-set of `paths` and fail in the middle - not "atomic" (to be addressed by another PR?)

* Only reporting the first encountered error (using the `?` operator)

Reply:
* Yes but we need to decide this behavior, I believe we can decide it when redefine the API on 2.0
* Looking at the API it can happened but looking at the impl it can not, again we should give an api that can expose such errors
* Yes but again this is an api limitation